CDTDatastore/touchdb/TDReplicator.m

```diff
@@ -800,10 +863,35 @@ - (void)saveLastSequence
     _lastSequenceChanged = _overdueForSave = NO;
 
     CDTLogInfo(CDTREPLICATION_LOG_CONTEXT, @"%@ checkpointing sequence=%@", self, _lastSequence);
-    NSMutableDictionary* body = [_remoteCheckpoint mutableCopy];
-    if (!body) body = $mdict();
-    [body setValue:_lastSequence forKey:@"lastSequence"];
-    
+    NSMutableDictionary* body = [self.remoteCheckpoint mutableCopy];
+
+    // upgrade the replication doc to the new format.
+    if (body && !body[@"history"]) {
+        NSMutableArray<NSDictionary<NSString*, NSObject*>*>* history = [@[ @{
+            @"session_id" : [[NSUUID UUID] UUIDString],
+            @"recorded_seq" : body[@"lastSequence"]
+        } ] mutableCopy];
+        body[@"history"] = history;
+        body[@"replication_id_version"] = @(3);
+        body[@"session_id"] = history[0][@"session_id"];
+        body[@"source_last_seq"] = body[@"lastSequence"];
+        body[@"lastSequence"] = nil;
```

Comment:
Ah I see the dictionary uses `nil` for removal I thought this was setting a `nil` value (but I see that is done differently) and then relying on not writing a JSON null to not get this in the output.